test/common/engine_test.cc

```diff
@@ -79,12 +80,58 @@ TEST_F(EngineTest, EarlyExit) {
                                    } /*on_exit*/,
                                    &test_context /*context*/};
 
-  engine_ = std::make_unique<EngineHandle>(callbacks, level);
+  engine_ = std::make_unique<EngineHandle>(callbacks, level, "");
   ASSERT_TRUE(test_context.on_engine_running.WaitForNotificationWithTimeout(absl::Seconds(3)));
 
   engine_.reset();
   ASSERT_TRUE(test_context.on_exit.WaitForNotificationWithTimeout(absl::Seconds(3)));
 
   start_stream(0, {}, false);
 }
+
+TEST_F(EngineTest, ComponentLogLevel) {
+  const std::string level = "debug";
+  const std::string component_log_level = "assert:info,misc:trace";
```

Comment:
the intent of this test as I assume is to verify engine will be running ok when the `component_log_level` is set correctly (or in the format that's expected), is this assumption right?

Reply:
I wrote this originally to verify that plumbing through the component log level without crashing (so yes your assumption is correct), though this test might be unnecessary now that we there are integration tests for Kotlin and iOS. The other test in here which validates that we don't crash when we get an invalid string is more valid.

Reply:
👌